### PR TITLE
feat(utils): amp contract states

### PR DESCRIPTION
**Pull request by sunzsolo**
## Describe your changes
- add Amplitude for pagination, contract states and contract details

### CODE REVIEW

File: src/lib/amplitude/track-event/trackContractStates.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { amp } from "../Amplitude";
+import type { AmpEvent } from "../types";
+
+export const trackContractStatesLoad = (
+  event:
+    | AmpEvent.USE_CONTRACT_STATES_LOAD_MORE
+    | AmpEvent.USE_CONTRACT_STATES_DOWNLOAD,
+  properties: {
+    current_states: number;
+    namespaces_count: number;
```
Comment (evilpeach) [inline src/lib/amplitude/track-event/trackContractStates.ts:10]: We should use camel case.

File: src/lib/pages/contract-details/components/contract-state/index.tsx [hunk 2]
```diff
@@ -98,8 +99,22 @@ export const ContractStates = ({ contractAddress }: ContractStatesProps) => {
           isLoading={isFetching || isFetchingNextPage}
           totalData={totalData}
           isCompleted={!hasNextPage}
-          onLoadMore={fetchNextPage}
-          onDownload={handleDownload}
+          onLoadMore={() => {
+            trackContractStatesLoad(AmpEvent.USE_CONTRACT_STATES_LOAD_MORE, {
+              current_states: totalData,
+              namespaces_count: namespaces.length,
+              namespaces,
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/contract-state/index.tsx:106]: Why are you returning `namespaces_count` even though you can obtain the length directly from `namespaces`?

File: src/lib/pages/contract-details/index.tsx [hunk 3]
```diff
@@ -304,14 +305,18 @@ const ContractDetails = observer(() => {
...
   ) as ContractAddr;
+  // TODO: fix assertion later
+  const tab = getFirstQueryParam(router.query.tab) as TabIndex;
+
   const contractData = useContractData(contractAddressParam);
 
   useEffect(() => {
-    if (router.isReady) track(AmpEvent.TO_CONTRACT_DETAIL);
-  }, [router.isReady]);
+    if (router.isReady && tab) track(AmpEvent.TO_CONTRACT_DETAIL, { tab });
```
Comment (evilpeach) [inline src/lib/pages/contract-details/index.tsx:318]: Do you intend to prevent triggering when `tab` value is null or undefined?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/pages/contract-details/index.tsx (hunk 3 above)
Comment (sunzsolo) [reply src/lib/pages/contract-details/index.tsx:318]: Yes

**Review by evilpeach — APPROVED** [review]

**Pull request merged**